app/src/main/java/com/mealsmadeeasy/model/MealPlanEntry.kt

```diff
@@ -5,5 +5,5 @@ import org.joda.time.DateTime
 data class MealPlanEntry(
         val date: DateTime,
         val mealPeriod: MealPeriod,
-        val meals: List<Meal>
+        var meals: List<Meal>
```

Comment:
Keep this as a val. I think that immutability is much more preferable than efficiency (it prevents a lot of crap with race conditions and unexpected modifications at the expense of more data transfer). When you actually want to modify a meal plan, you can call `mealPlanEntry.copy(meals = mealPlanEntry.meals + someListOfMeals)` as an example. Copy has all the same parameters as the constructor, but they default to the current value so they're all optional.

Reply:
The BehaviorSubject array is immutable too, though, so unless I'm missing something, you won't be able to assign the MealPlanEntry to the updated one with the changed list.